mycpp/gc_str.cc

```diff
@@ -525,6 +525,13 @@ List<Str*>* Str::split(Str* sep) {
   return this->split(sep, len(this));
 }
 
+int Str::hash(HashFunc h) {
+  if ((hash_value_ & Str::kHashFlagMask) == 0) {
+    hash_value_ = h(data_, len_) | Str::kHashFlagMask;
```

Comment:
How about some unit tests for this function?

Maybe it's the same thing but it seems more logical / conventional to do `hash & 0x7fff_ffff` to mask off one bit

Also I would just inline the constant if it's only used in one place ... it makes it harder to read the formula if it's scattered

Reply:
In this case we actually want to _set_ the bit. So, it's correct as-is. I agree that inlining will be less confusing, though. Will do.

The line immediately below this just shifts the hash to the right 1 bit to lop the flag bit off. Short and sweet.

Reply:
Will add a test 👍